src/main/java/pl/edu/icm/sparkutils/avro/SparkAvroLoader.java

```diff
@@ -49,8 +50,10 @@ private SparkAvroLoader() {
         @SuppressWarnings("unchecked")
         JavaPairRDD<AvroKey<T>, NullWritable> inputRecords = (JavaPairRDD<AvroKey<T>, NullWritable>)
                 sc.newAPIHadoopFile(avroDatastorePath, AvroKeyInputFormat.class, avroRecordClass, NullWritable.class, job.getConfiguration());
+        
+        
 
-        JavaRDD<T> input = inputRecords.map(tuple -> tuple._1.datum());
+        JavaRDD<T> input = inputRecords.map(tuple -> AvroUtils.cloneAvroRecord(tuple._1.datum()));
```

Comment:
A very clever move ;-)
But it needs explanation, a comment here above the line and above the method.